src/subhalo_tracking.cpp

```diff
@@ -853,28 +869,46 @@ class SubhaloMasker_t
   {
     ExclusionList.reserve(np_guess);
   }
-  void Mask(HBTInt subid, vector<Subhalo_t> &Subhalos)
+  void Mask(HBTInt subid, vector<Subhalo_t> &Subhalos, int SnapshotIndex)
   {
     auto &subhalo = Subhalos[subid];
     for (auto nestedid :
          subhalo
            .NestedSubhalos) // TODO: do we have to do it recursively? satellites are already masked among themselves?
-      Mask(nestedid, Subhalos);
+      Mask(nestedid, Subhalos, SnapshotIndex);
 
     if (subhalo.Nbound <= 1)
       return; // skip orphans
 
+    if (subhalo.SnapshotIndexOfBirth == SnapshotIndex)
```

Comment:
Aren't you able to do the same by checking whether subhalo.TrackId == -1?

Reply:
What would happen to an orphan that find themselves in a FOF group without a bound component, and are hence allowed to re-grow? i.e. it is not "new", but it would effectively undergo the same feeding process.

Reply:
I think that's the situation where my crash happened. But if the orphan has the correct type of tracer it would just get fed the particles. Isn't that ok? The FoF group would have to be barely resolved and any orphan was likely generated by the same group temporarily dropping below the resolution limit earlier on, so I think feeding the existing orphan is better than generating a new track.

Reply:
There might be a problem if we have a zero particle orphan at this point.

Reply:
I don't think this is a issue. If we regrow a subhalo then I think we should definitely have tracers, because the FoF group itself will have tracers. That should mean we won't ever need to add back a tracer after masking.

Reply:
In FeedCentrals the line
```
auto mostbndid = Host.Particles[0].Id;
```
will be an out of bounds reference if the subhalo that was swapped with the host was a zero particle orphan. And while the FoF will contain tracers they're not sorted by binding energy so we have no way to pick which one to use.

But maybe this is better dealt with in #23?

Reply:
Good catch. I was getting in that branch during FeedCentrals, so that might be it!